__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -1,34 +1,173 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable global-require */
+/* eslint-disable react/display-name */
 import * as React from 'react';
-import {render} from '@testing-library/react-native';
+import FastImage from 'react-native-fast-image';
+import {act, fireEvent, render} from '@testing-library/react-native';
 
 import ChatItemMyTextV2 from '../../../../src/components/AnonymousChat/child/ChatItemMyTextV2';
+import {Context} from '../../../../src/context';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../../src/utils/constants';
 
-describe('TESTING ChatItemTargetText', () => {
-  it('RENDER not continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar="https://www.google.com"
-        username="username"
-        isContinuous={false}
-        message="message"
-        time="time"
-      />
+jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});
+jest.mock('react-native-reanimated', () => ({
+  ...jest.requireActual('react-native-reanimated'),
+  useSharedValue: jest.fn(() => ({value: 0})),
+  useAnimatedStyle: jest.fn(() => ({})),
+  View: require('react-native/Libraries/Components/View/View').View,
+  Animated: {
+    View: require('react-native/Libraries/Components/View/View').View
+  }
+}));
+jest.mock('react-native-gesture-handler', () => {
+  const originalModule = jest.requireActual('react-native-gesture-handler');
+  const {View} = require('react-native');
+  const React = require('react');
+
+  return {
+    ...originalModule,
+    Swipeable: React.forwardRef(({children, ...props}, ref) => {
+      React.useImperativeHandle(ref, () => ({
+        close: jest.fn()
+      }));
+
+      return <View {...props}>{children}</View>;
+    })
+  };
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe('TESTING ChatItemMyTextV2', () => {
+  const avatar = (
+    <FastImage
+      style={{height: 24, width: 24, borderRadius: 12}}
+      source={{uri: DEFAULT_PROFILE_PIC_PATH}}
+    />
+  );
+
+  it('should call setReplyPreview when swiped', () => {
+    const setReplyPreview = jest.fn();
+    const contextValue = {
+      chat: [{replyTarget: null}, setReplyPreview]
+    };
+
+    const {getByTestId} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
+    );
+
+    act(() => {
+      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    });
+
+    expect(setReplyPreview).toHaveBeenCalled();
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test 'should call setReplyPreview when swiped' seems to be testing the swipe functionality of the `Swipeable` component. However, since the `Swipeable` component is being mocked and the mock does not simulate any swipe behavior, this test may not accurately reflect the actual behavior of the component in a real-world scenario. Consider using a library like `@testing-library/user-event` to simulate more complex user interactions.



<!-- This is an auto-generated comment by OSS CodeRabbit -->